### ISSUE TITLE
Better error message when a Material isn't in a TextField's ancestry

### ISSUE THREAD

**Issue by pakivika** [flutter/flutter#14764]
## Steps to Reproduce

Change cupertino_navigation_demo.dart in flutter_gallery as follows (see code snippet)
- Change CupertinoDemoTab2 to Stateful
- In the state, add TextFormField to the widget tree
- Run app in iPhone (slow mode)
- Go to Cupertino Navigation Demo and Tab2, then insert some text to the TextFormField
- Switch to another tab and come back to Tab2
- Try to insert text again into TextFormField

Expected result:
- TextFormField gets focus and you can insert/modify text

Actual result:
- TextFormField sometimes doesn't get focus. Inkwell seems to create visual effect but then fails.
- Note! Doesn't always reproduce

## Code snippet

```
flutter_gallery.../lib/demo/cupertino/cupertino_navigation_demo.dart
...
class CupertinoDemoTab2 extends StatefulWidget {

  @override
  CupertinoDemoTab2State createState() => new CupertinoDemoTab2State();
}

class CupertinoDemoTab2State extends State<CupertinoDemoTab2> {
  String msg;

  @override
  Widget build(BuildContext context) {
    return new CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar(
        middle: const Text('Support Chat'),
        trailing: const ExitButton(),
      ),
      child: new ListView(
        children: <Widget>[
          new Tab2Header(),
        ]..addAll(buildTab2Conversation())
          ..add(
            new TextFormField(
              keyboardType: TextInputType.text,
              style: Theme.of(context).textTheme.title,
              onFieldSubmitted: (String value) { msg = value; },
              decoration: const InputDecoration(
                icon: const Icon(Icons.comment),
                hintText: 'Add free-text comments here',
              ),
            ),
          ),
      ),
    );
  }
}
...
```

## Logs

```
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following assertion was thrown while handling a gesture:
'package:flutter/src/material/ink_well.dart': Failed assertion: line 41 pos 15: 'controller !=
null': is not true.

Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new

When the exception was thrown, this was the stack:
#2      new InteractiveInkFeature (package:flutter/src/material/ink_well.dart:41:15)
#3      new InkSplash (package:flutter/src/material/ink_splash.dart:130:8)
#4      _InkSplashFactory.create (package:flutter/src/material/ink_splash.dart:61:16)
#5      _TextFieldState._createInkFeature (package:flutter/src/material/text_field.dart:366:46)
#6      _TextFieldState._startSplash (package:flutter/src/material/text_field.dart:405:42)
#7      _TextFieldState._handleTapDown (package:flutter/src/material/text_field.dart:384:5)
#8      TapGestureRecognizer._checkDown.<anonymous closure> (package:flutter/src/gestures/tap.dart:142:57)
#9      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102:24)
#10     TapGestureRecognizer._checkDown (package:flutter/src/gestures/tap.dart:142:9)
#11     TapGestureRecognizer.didExceedDeadline (package:flutter/src/gestures/tap.dart:114:5)
#13     _Timer._runTimers (dart:isolate-patch/dart:isolate/timer_impl.dart:367)
#14     _Timer._handleMessage (dart:isolate-patch/dart:isolate/timer_impl.dart:401)
#15     _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:164)
(elided 3 frames from class _AssertionError and package dart:async-patch)

Handler: onTapDown
Recognizer:
  TapGestureRecognizer#91c17(debugOwner: GestureDetector, state: possible)
════════════════════════════════════════════════════════════════════════════════════════════════════
Another exception was thrown: 'package:flutter/src/material/ink_well.dart': Failed assertion: line 41 pos 15: 'controller != null': is not true.
Another exception was thrown: 'package:flutter/src/material/ink_well.dart': Failed assertion: line 41 pos 15: 'controller != null': is not true.
```

## Flutter Doctor

```
$ flutter doctor -v
[  +16 ms] [/usr/local/bin/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +35 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/alpha
[        ] [/usr/local/bin/flutter/] git rev-parse --abbrev-ref HEAD
[  +10 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] alpha
[        ] [/usr/local/bin/flutter/] git ls-remote --get-url origin
[  +14 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[        ] [/usr/local/bin/flutter/] git log -n 1 --pretty=format:%H
[  +12 ms] Exit code 0 from: git log -n 1 --pretty=format:%H
[        ] 2e449f06f0a3be076e336ad6b30b0e9ec99dbdfe
[        ] [/usr/local/bin/flutter/] git log -n 1 --pretty=format:%ar
[  +11 ms] Exit code 0 from: git log -n 1 --pretty=format:%ar
[        ] 3 weeks ago
[   +1 ms] [/usr/local/bin/flutter/] git describe --match v*.*.* --first-parent --long --tags
[  +14 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v0.0.21-0-g2e449f06f
[ +432 ms] /usr/bin/defaults read /Applications/Android Studio.app/Contents/Info CFBundleShortVersionString
[  +76 ms] Exit code 0 from: /usr/bin/defaults read /Applications/Android Studio.app/Contents/Info CFBundleShortVersionString
[        ] 3.0
[ +197 ms] [✓] Flutter (on Mac OS X 10.13.3 17D47, locale en-FI, channel alpha)
[   +1 ms]     • Flutter version 0.0.21 at /usr/local/bin/flutter
[        ]     • Framework revision 2e449f06f0 (3 weeks ago), 2018-01-29 14:26:51 -0800
[        ]     • Engine revision 6921873c71
[        ]     • Tools Dart version 2.0.0-dev.16.0
[        ]     • Engine Dart version 2.0.0-edge.da1f52592ef73fe3afa485385cb995b9aec0181a
[ +189 ms] /usr/bin/defaults read /Applications/Android Studio.app/Contents/Info CFBundleShortVersionString
[  +83 ms] Exit code 0 from: /usr/bin/defaults read /Applications/Android Studio.app/Contents/Info CFBundleShortVersionString
[        ] 3.0
[  +98 ms] java -version
[ +130 ms] [✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[        ]     • Android SDK at /Users/pakivika/Library/Android/sdk
[        ]     • Android NDK location not configured (optional; useful for native profiling support)
[        ]     • Platform android-27, build-tools 27.0.3
[        ]     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
[        ]     • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
[ +190 ms] DevToolsSecurity -status
[  +51 ms] Developer mode is currently enabled.
[        ] python -c import six
[  +68 ms] idevice_id -h
[   +8 ms] idevice_id -h
[   +6 ms] idevice_id -l
[  +17 ms] c992803dfb58d4b89db1f0b0ecba2eeb63b9acab
[        ] idevicename
[ +139 ms] ideviceinstaller -h
[   +8 ms] ios-deploy --version
[  +14 ms] ios-deploy --version
[  +14 ms] 1.9.2
[   +1 ms] ios-deploy --version
[  +32 ms] ios-deploy --version
[  +14 ms] 1.9.2
[   +3 ms] pod --version
[ +409 ms] pod --version
[ +410 ms] 1.3.1
[   +2 ms] pod --version
[ +401 ms] 1.3.1
[        ] [✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
[        ]     • Xcode at /Applications/Xcode.app/Contents/Developer
[        ]     • Xcode 9.2, Build version 9C40b
[        ]     • ios-deploy 1.9.2
[        ]     • CocoaPods version 1.3.1
[   +1 ms] [✓] Android Studio (version 3.0)
[        ]     • Android Studio at /Applications/Android Studio.app/Contents
[        ]     • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
[   +1 ms] /usr/bin/defaults read /Applications/IntelliJ IDEA CE.app/Contents/Info CFBundleShortVersionString
[  +88 ms] Exit code 0 from: /usr/bin/defaults read /Applications/IntelliJ IDEA CE.app/Contents/Info CFBundleShortVersionString
[        ] 2017.3
[  +74 ms] [✓] IntelliJ IDEA Community Edition (version 2017.3)
[        ]     • Flutter plugin version 20.0.3
[        ]     • Dart plugin version 173.3727.108
[   +4 ms] /Users/pakivika/Library/Android/sdk/platform-tools/adb devices -l
[  +16 ms] Exit code 0 from: /Users/pakivika/Library/Android/sdk/platform-tools/adb devices -l
[        ] List of devices attached
[   +5 ms] idevice_id -h
[  +58 ms] which ideviceinstaller
[   +6 ms] Exit code 0 from: which ideviceinstaller
[        ] /usr/local/bin/ideviceinstaller
[        ] which iproxy
[   +7 ms] Exit code 0 from: which iproxy
[        ] /usr/local/bin/iproxy
[   +5 ms] /usr/bin/xcrun simctl list --json devices
[ +176 ms] [✓] Connected devices
[        ]     • Pasi's iPhone • xxx • ios • iOS 11.2.5
[  +15 ms] "flutter doctor" took 3,383ms.
[  +64 ms] ensureAnalyticsSent: 62ms
[   +2 ms] exiting with code 0
```

**Comment by najeira** [comment]
similar to #12397

**Comment by cbracken** [comment]
@xster does the fix you landed to #12397 fix this as well?
